pkg/manager/internal.go

```diff
@@ -552,13 +550,16 @@ func (cm *controllerManager) engageStopProcedure(stopComplete <-chan struct{}) e
 	defer cm.recorderProvider.Stop(cm.shutdownCtx)
 	defer func() {
 		// Cancel leader election only after we waited. It will os.Exit() the app for safety.
-		if cm.resourceLock != nil {
-			// After asking the context to be cancelled, make sure
-			// we wait for the leader stopped channel to be closed, otherwise
-			// we might encounter race conditions between this code
-			// and the event recorder, which is used within leader election code.
+		// After asking the context to be cancelled, make sure
+		// we wait for the leader stopped channel to be closed, otherwise
+		// we might encounter race conditions between this code
+		// and the event recorder, which is used within leader election code.
+		if cm.leaderElectionCancel != nil {
```

Comment:
It seems you only have to check one of `cm.leaderElectionCancel` and `cm.leaderElectionStopped` for now they are assigned at the same place.

Reply:
Yes but I think maybe checking them individually is safer 😄